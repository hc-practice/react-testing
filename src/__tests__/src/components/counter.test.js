import React from "react";
import renderer from "react-test-renderer";
import Counter from "../../../components/counter/counter.js";

describe("<Counter />", () => {
  it("it is alive", () => {
    expect(true).toBeTruthy();
  });

  it("proof of life", () => {
    let component = shallow(<Counter />);
    expect(component.find("section").exists()).toBeTruthy();
  });

  it("changes state on click to 1", () => {
    let component = mount(<Counter />);
    let clickerUp = component.find("a.upClick");
    clickerUp.simulate("click");
    expect(component.state("count")).toBe(1);
    expect(component.find("span").text()).toContain(1);
  });

  it("changes state on click to -1", () => {
    let component = mount(<Counter />);
    let clickerDown = component.find("a.downClick");
    clickerDown.simulate("click");
    expect(component.state("count")).toBe(-1);
    expect(component.find("span").text()).toContain(-1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
