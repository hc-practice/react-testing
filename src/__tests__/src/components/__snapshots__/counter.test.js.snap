// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<section
  className="counter"
>
  <a
    className="downClick"
    href="#"
    onClick={[Function]}
  >
    &lt;&lt;
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="upClick"
    href="#"
    onClick={[Function]}
  >
    &gt;&gt;
  </a>
</section>
`;
